### PR TITLE
feat: [UIE-6576] - Dbaas total disk size and used disk size

### PR DESCRIPTION
**Pull request by corya-akamai**
## Description 📝
Adding two new fields to display (1) the total disk size available for the database (Selected size - OS overhead) and (2) the used disk space so the customer will now how much space is remaining in the DB. BE will be feature gated, so UI will fallback to current storage field if the API response does not contain the total_disk_space

## Major Changes 🔄
- Field total_disk_space_gb
- Field used_disk_space_gb
- database_low_disk_space_remaining event

## Preview 📷
Screenshot 2023-08-02 at 1 53 30 PM

## How to test 🧪
1. Use the mock service workers
2. Navigate to http://localhost:3000/databases/mysql/0
3. Verify Total Disk Size and Used fields are present. Modify packages > manager > src > factories > databases.ts > total_disk_size_gb to undefined to revert to previous view with the Storage field.
4. E2E test will be added from QA sujai

### CODE REVIEW

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryClusterConfiguration.tsx [hunk 1]
```diff
@@ -65,6 +66,14 @@ export const DatabaseSummaryClusterConfiguration = (props: Props) => {
       ? 'Primary'
       : `Primary +${database.cluster_size - 1} replicas`;
 
+  const sxTooltipIcon = {
+    padding: '0px',
+    marginLeft: '4px',
+  };
+
+  const storageCopy =
```
Comment (abailly-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryClusterConfiguration.tsx:74]:
```suggestion
  const STORAGE_COPY =
```

File: packages/manager/src/dev-tools/FeatureFlagTool.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ const MOCK_FEATURE_FLAGS_STORAGE_KEY = 'devTools/mock-feature-flags';
 const options: { flag: keyof Flags; label: string }[] = [
   { flag: 'metadata', label: 'Metadata' },
   { flag: 'databaseBeta', label: 'Database Beta' },
+  { flag: 'databases', label: 'Databases' },
```
Comment (abailly-akamai) [inline packages/manager/src/dev-tools/FeatureFlagTool.tsx:16]: This flag naming isn't descriptive enough. Let's make sure we know what it's for

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryClusterConfiguration.tsx [hunk 3]
```diff
@@ -102,10 +111,28 @@ export const DatabaseSummaryClusterConfiguration = (props: Props) => {
           <Typography className={classes.label}>CPUs</Typography>
           {type.vcpus}
         </Box>
-        <Box display="flex">
-          <Typography className={classes.label}>Storage</Typography>
-          {convertMegabytesTo(type.disk, true)}
-        </Box>
+        {database.total_disk_size_gb ? (
```
Comment (abailly-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryClusterConfiguration.tsx:114]: Don't we want to make a new feature flag and use it as an additional conditional here? This works but wondering if the scope may expand?

File: packages/api-v4/src/databases/types.ts [hunk 4]
```diff
@@ -128,6 +128,8 @@ export interface BaseDatabase {
   hosts: DatabaseHosts;
   port: number;
   updates: UpdatesSchedule;
+  total_disk_size_gb?: number;
+  used_disk_size_gb?: number;
```
Comment (abailly-akamai) [inline packages/api-v4/src/databases/types.ts:132]: Are those values always going to return from the API? if so they shouldn't be optional (and we can use the feature flag to render the new data conditionally)

**Review by abailly-akamai — COMMENTED** [review]
Thanks for submitting! left a few comments

**Review by bnussman-akamai — APPROVED** [review]
Looks good in alpha! 🎉

I'd like to know why this is feature flagged on the API side. Do we expect this feature to ever be turned off? Should we add our own feature flag on the frontend for this?

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryClusterConfiguration.tsx (hunk 3 above)
Comment (corya-akamai) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryClusterConfiguration.tsx:114]: The backend has a feature gate for the rollout. I asked about a UI flag, but they said the backend API response should suffice.

**Comment by corya-akamai** [comment]
> Looks good in alpha! 🎉
> 
> I'd like to know why this is feature flagged on the API side. Do we expect this feature to ever be turned off? Should we add our own feature flag on the frontend for this?

The backend added a feature flag for the rollout. It's turned off initially to verify no regressions were introduced. When I asked about adding a flag on the UI, they said the API response can drive the behavior.

**Review by abailly-akamai — APPROVED** [review]
Tested in alpha and looks good. 🚢

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryClusterConfiguration.tsx (hunk 3 above)
Comment (abailly-akamai) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryClusterConfiguration.tsx:114]: gotcha, thx for the explanation

**Comment by bnussman-akamai** [comment]
Ran e2es locally and everything passed ✅

**Pull request merged**